12-offer-encoding.md

```diff
@@ -238,296 +233,328 @@ The human-readable prefix for offers is `lno`.
 
 1. `tlv_stream`: `invoice`
 2. types:
-    1. type: 3 (`chain`)
+    1. type: 0 (`invoice_request_payer_info`)
     2. data:
-        * [`chain_hash`:`chain`]
-    1. type: 4 (`offer_id`)
+        * [`...*byte`:`blob`]
+    1. type: 2 (`offer_chains`)
     2. data:
-        * [`sha256`:`offer_id`]
-    1. type: 8 (`amount`)
+        * [`...*chain_hash`:`chains`]
+    1. type: 4 (`offer_metadata`)
     2. data:
-        * [`tu64`:`msat`]
-    1. type: 10 (`description`)
+        * [`...*byte`:`data`]
+    1. type: 6 (`offer_currency`)
+    2. data:
+        * [`...*utf8`:`iso4217`]
+    1. type: 8 (`offer_amount`)
+    2. data:
+        * [`tu64`:`amount`]
+    1. type: 10 (`offer_description`)
     2. data:
         * [`...*utf8`:`description`]
-    1. type: 12 (`features`)
+    1. type: 12 (`offer_features`)
     2. data:
         * [`...*byte`:`features`]
-    1. type: 16 (`paths`)
+    1. type: 14 (`offer_absolute_expiry`)
     2. data:
-        * [`...*blinded_path`:`paths`]
-    1. type: 18 (`blindedpay`)
-    2. data:
-        * [`...*blinded_payinfo`:`payinfo`]
-    1. type: 19 (`blinded_capacities`)
+        * [`tu64`:`seconds_from_epoch`]
+    1. type: 16 (`offer_paths`)
     2. data:
-        * [`...*u64`:`incoming_msat`]
-    1. type: 20 (`issuer`)
+        * [`...*blinded_path`:`paths`]
+    1. type: 18 (`offer_issuer`)
     2. data:
         * [`...*utf8`:`issuer`]
-    1. type: 30 (`node_id`)
+    1. type: 20 (`offer_quantity_min`)
+        * [`tu64`:`min`]
+    1. type: 22 (`offer_quantity_max`)
+    2. data:
+        * [`tu64`:`max`]
+    1. type: 24 (`offer_node_id`)
     2. data:
         * [`point`:`node_id`]
-    1. type: 32 (`quantity`)
+    1. type: 26 (`offer_send_invoice`)
+    1. type: 80 (`invoice_request_chain`)
     2. data:
-        * [`tu64`:`quantity`]
-    1. type: 34 (`refund_for`)
+        * [`chain_hash`:`chain`]
+    1. type: 82 (`invoice_request_amount`)
     2. data:
-        * [`sha256`:`refunded_payment_hash`]
-    1. type: 38 (`payer_key`)
+        * [`tu64`:`msat`]
+    1. type: 84 (`invoice_request_features`)
+    2. data:
+        * [`...*byte`:`features`]
+    1. type: 86 (`invoice_request_quantity`)
+    2. data:
+        * [`tu64`:`quantity`]
+    1. type: 88 (`invoice_request_payer_key`)
     2. data:
         * [`point`:`key`]
-    1. type: 39 (`payer_note`)
+    1. type: 89 (`invoice_request_payer_note`)
     2. data:
         * [`...*utf8`:`note`]
-    1. type: 40 (`created_at`)
+    1. type: 160 (`invoice_paths`)
+    2. data:
+        * [`...*blinded_path`:`paths`]
+    1. type: 162 (`invoice_blindedpay`)
+    2. data:
+        * [`...*blinded_payinfo`:`payinfo`]
+    1. type: 164 (`invoice_created_at`)
     2. data:
         * [`tu64`:`timestamp`]
-    1. type: 42 (`payment_hash`)
+    1. type: 168 (`invoice_payment_hash`)
     2. data:
         * [`sha256`:`payment_hash`]
-    1. type: 44 (`relative_expiry`)
+    1. type: 170 (`invoice_relative_expiry`)
```

Comment:
Could we move this immediately after `invoice_created_at`? Seems more natural there.

Reply:
OK, shuffled that around, used 166 which was omitted somehow.